src/main/java/org/csanchez/jenkins/plugins/kubernetes/ProvisioningCallback.java

```diff
@@ -145,73 +147,44 @@ public Node call() throws Exception {
             String podId = pod.getMetadata().getName();
             String namespace = Strings.isNullOrEmpty(t.getNamespace()) ? client.getNamespace() : t.getNamespace();
 
-            LOGGER.log(Level.FINE, "Creating Pod: {0} in namespace {1}", new Object[] { podId, namespace });
+            String[] logArgs = new String[] { namespace, podId };
+            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", logArgs);
             pod = client.pods().inNamespace(namespace).create(pod);
-            LOGGER.log(Level.INFO, "Created Pod: {0} in namespace {1}", new Object[] { podId, namespace });
 
-            // We need the pod to be running and connected before returning
-            // otherwise this method keeps being called multiple times
-            ImmutableList<String> validStates = ImmutableList.of("Running");
-
-            int i = 0;
-            int j = 100; // wait 600 seconds
-
-            List<ContainerStatus> containerStatuses = null;
-
-            // wait for Pod to be running
-            for (; i < j; i++) {
-                LOGGER.log(Level.INFO, "Waiting for Pod to be scheduled ({1}/{2}): {0}", new Object[] { podId, i, j });
-                Thread.sleep(6000);
-                pod = cloud.connect().pods().inNamespace(namespace).withName(podId).get();
-                if (pod == null) {
-                    throw new IllegalStateException("Pod no longer exists: " + podId);
-                }
-
-                containerStatuses = pod.getStatus().getContainerStatuses();
-                List<ContainerStatus> terminatedContainers = new ArrayList<>();
-                Boolean allContainersAreReady = true;
-                for (ContainerStatus info : containerStatuses) {
-                    if (info != null) {
-                        if (info.getState().getWaiting() != null) {
-                            // Pod is waiting for some reason
-                            LOGGER.log(Level.INFO, "Container is waiting {0} [{2}]: {1}",
-                                    new Object[] { podId, info.getState().getWaiting(), info.getName() });
-                            // break;
-                        }
-                        if (info.getState().getTerminated() != null) {
-                            terminatedContainers.add(info);
-                        } else if (!info.getReady()) {
-                            allContainersAreReady = false;
+            LOGGER.log(Level.INFO, "Waiting for pod to be ready: {0}/{1}", logArgs);
+            try {
+                client.pods().inNamespace(namespace).withName(podId).waitUntilReady(10, TimeUnit.MINUTES);
```

Comment:
Wouldn't this be a good opportunity to make this wait time configurable, at least as a system property?

10 minutes seems a bit log for starting up a slave pod (especially if you want to manually test the error handling code 😁 ).

Reply:
I think it was 10 minutes already

Reply:
needs to be in MILLISECONDS due to a bug in the k8s client